hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -931,17 +940,22 @@ private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> lookupSecondaryKe
       List<String> sortedSecondaryKeys = new ArrayList<>(secondaryKeys);
       secondaryKeySet.addAll(sortedSecondaryKeys);
       Collections.sort(sortedSecondaryKeys);
+      Set<String> deletedRecordKeysFromLogs = new HashSet<>();
 
       logRecordScanner.getRecords().forEach(record -> {
         HoodieMetadataPayload payload = record.getData();
-        String secondaryKey = payload.key;
-        if (secondaryKeySet.contains(secondaryKey)) {
-          String recordKey = payload.getRecordKeyFromSecondaryIndex();
-          logRecordsMap.computeIfAbsent(secondaryKey, k -> new HashMap<>()).put(recordKey, record);
+        if (!payload.isDeleted()) {
+          String secondaryKey = payload.key;
```

Comment:
Can we follow the deletion style of `HoodieBackedTableMetadata.readFromBaseAndMergeWithLogRecords` to make the merging mode concise.

Reply:
hey @codope : can you respond to Danny's Q here.

Reply:
Yes, this can be refactored. We need to move this logic inside `HoodieMetadataPayload.combineSecondaryIndexRecord`, and need to override `MetadataPartitionType.combineMetadataPayloads` for secondary index. Followup - HUDI-8436